### ISSUE TITLE
ansible-lint should not work on temp files

### ISSUE THREAD

**Issue by languitar** [dense-analysis/ale#1814]
`ansible-lint` (at least in version 2.6) reports less warnings on files that are outside of a usual ansible project:

```
languitar@jaco ~/src/server-configuration (master)> ansible-lint -p site.yml 
/home/languitar/src/server-configuration/roles/mailserver/tasks/main.yml:272: [EANSIBLE0011] All tasks should be named
languitar@jaco ~/src/server-configuration (master) [2]> mkdir /tmp/asdfasdf
languitar@jaco ~/src/server-configuration (master)> cp site.yml /tmp/asdfasdf/site.yml
languitar@jaco ~/src/server-configuration (master)> ansible-lint -p /tmp/asdfasdf/site.yml 
```
Thus, the ansible-lint configuration for ALE should not use temp files.

**Comment by languitar** [comment]
Hm, actually, ansible-lint always needs to be run on the playbook, not an arbitrary role file:

```
languitar@jaco ~/src/server-configuration (master)> ansible-lint -h
Usage: ansible-lint [options] playbook.yml [playbook2 ...]
```

**Comment by w0rp** [comment]
What would be best would be setting up two linter definitions, one that shows more errors and warnings and can only be run when you save a file, and another which shows the errors and warnings that can be found without having to save a file.

**Comment by languitar** [comment]
But I think my second comment is most important. I don't get a warning in `/home/languitar/src/server-configuration/roles/mailserver/tasks/main.yml` if I pass this file to the linter. It will only report warnings if I specify the playbook file (`site.yml`) that eventually includes that role. Thus, it never makes sense to lint anything else than playbooks, even if you want to get warnings in roles.

**Comment by w0rp** [comment]
It's nice to be able to get _some_ feedback _before_ you save a file. If you only run the llinter against the playbook and nothing else, you won't be able to see that problems have been fixed until you save a file.

**Comment by languitar** [comment]
Ok. But how does it work out with ale to get linter errors in roles files then? Because if you edit a role file, you would actually have to pass a playbook file using this role to the linter to get feedback there.

**Comment by w0rp** [comment]
I suggest writing two linter definitions, as I said before. One to check a temporary file, and another to check the whole playbook.

**Comment by languitar** [comment]
I see, but how is the second linter triggered in case I edit a role file? And how does one know which playbook file to use as the argument?

**Comment by timblaktu** [comment]
It seems to me that this problem is analogous to a problem with pylint, which @w0rp discusses [here](https://github.com/dense-analysis/ale/issues/459#issuecomment-292937216). I believe pylint support in ALE has been such that it really only lints on file save, and I would suggest that for the same reasons, the ansible-lint'er should behave the same way. 

This is not a problem in general, since yamllint can still lint on-the-fly for you.

**Comment by varac** [comment]
I'd also love to have proper ansible-lint support for roles in ale !

**Comment by w0rp** [comment]
I think someone needs to change the linter to run against the file on disk, instead of as you type, as ansible-lint doesn't offer options for supporting anything else.

**Comment by varac** [comment]
Anybody ? Doesn't sound too hard for somebody familiar with developing ALE/plugins, I'm not at all except for using it...

**Comment by hsanson** [comment]
This may be solved with #3601
